app/components/checkout/braintree/BraintreePayments.tsx

```diff
@@ -0,0 +1,108 @@
+import { addPaymentToOrder } from '~/providers/checkout/checkout';
+import { CurrencyCode } from '~/generated/graphql';
+
+export function BraintreeDropIn(props: { show: boolean, authorization: string, fullAmount: number, currencyCode : CurrencyCode }) {
+    const { show, authorization, fullAmount, currencyCode } = props;
+
+    const [braintreeInstance, setBraintreeInstance] = useState<Dropin>();
+    const [enablePaymentButton, setEnablePaymentButton] = useState<boolean>();
+    const [processing, setProcessing] = useState<boolean>(false);
+
+    const submit = useSubmit();
+
+    const submitPayment = async () => {
+        setProcessing(true);
+        if (braintreeInstance) {
+            try{
+                const result = await braintreeInstance.requestPaymentMethod();
+
+                const formData = new FormData();
+                formData.set("paymentMethodCode", "braintree");
+                formData.set("paymentNonce", result.nonce);
+    
+                let request: Request;
+                request = new Request("");
+    
+                await addPaymentToOrder({ method: "braintree", metadata: result }, { request });
+    
+                submit(formData, { method: "post" });
+            }catch(e){
+                alert(e);
+                setProcessing(false);
+            } 
+        }
+    };
+
+    useEffect(() => {
+        if (show) {
+            const initializeBraintree = () => dropin.create({
+                authorization: authorization,
+                container: '#braintree-drop-in-div',
+                paypal: {
+                    flow: 'checkout',
+                    amount: fullAmount,
+                    currency: currencyCode.toString(),
+                }
+            }, function (error, instance) {
+                if (error)
+                    console.error(error)
+                else if (instance != null) {
+                    setBraintreeInstance(instance);
+                    instance.on("paymentMethodRequestable", (payload) => {
+                        setEnablePaymentButton(true);
+                    });
+                    instance.on("noPaymentMethodRequestable", () => {
+                        setEnablePaymentButton(false);
+                    });
+                }
+            });
+
+            if (braintreeInstance) {
+                braintreeInstance
+                    .teardown()
+                    .then(() => {
+                        initializeBraintree();
+                    });
+            } else {
+                initializeBraintree();
+            }
+        }
+    }, [show])
+
+    return (
+        <div
+            style={{ display: `${show ? "block" : "none"}` }}
+            className={"w-full h-full"}
+        >
+            <div
+                id={"braintree-drop-in-div"}
+            />
+
+            <input
+                type="hidden"
+                name="paymentMethodCode"
+                value={"braintree"}
```

Comment:
Not a huge deal, but for string literals you don't need to create a binding, so it can just be `value="braintree"`, `className="w-full h-full"` etc.